semeio/jobs/ots/ots.py

```diff
@@ -0,0 +1,477 @@
+                    pos=r2,
+                    youngs_modulus=self._youngs_modulus,
+                    poisson_ratio=self._poisson_ratio,
+                    seabed=self._seabed,
+                )
+                surf_displacement[idx_vtg, point] = dz2 - dz1
+
+        for pair in vintage_pairs:
+            self._report(method_name, pair[0], pair[1], len(points_to_calculate))
+
+            base_index = vintages.date.index(pair[0])
+            monitor_index = vintages.date.index(pair[1])
+            ts = -self._r_factor * (
+                surf_displacement[monitor_index] - surf_displacement[base_index]
+            )
+            self._divide_negative_shift(ts)
+
+            ts = ts * self._convention
+
+            ts_surfaces.append(self._create_surface(ts))
+
+        return ts_surfaces
+
+    def geertsma_ts(self, vintage_pairs):
+        """
+        Calculates TS using velocity. Slow.
+
+        :param vintage_pairs:
+        """
+
+        if len(vintage_pairs) < 1:
+            return 0, []
+
+        surface = self._surface
+        points_to_calculate = self._get_non_nan_points()
+
+        ts_surfaces = []
+        _, nz = surface.z3d.shape
+
+        for pair in vintage_pairs:
+            surf_displacement = np.zeros(len(surface))
+            logging.info(
+                "{:%x %X} TS: Calculating vintage"
+                " {:%Y.%m.%d} - {:%Y.%m.%d}".format(dt.now(), pair[0], pair[1])
+            )
+
+            self.add_survey("base", pair[1])
+            self.add_survey("monitor", pair[0])
+
+            for point in points_to_calculate:
+                for iz in range(nz):
+                    rz = surface.z3d[point, iz] - self._seabed
+                    if 0 <= rz <= (surface.z[point] - self._seabed):
+                        r1 = (surface.x[point], surface.y[point], rz)
+                        r2 = (surface.x[point], surface.y[point], rz + 0.1)
+                        # subsidence and displacement have opposite sign
+                        # should have minus here, more efficient
+                        # when calculating ts
+                        dz1 = self.subsidence.evalGeertsma(
+                            base_survey="base",
+                            monitor_survey="monitor",
+                            pos=r1,
+                            youngs_modulus=self._youngs_modulus,
+                            poisson_ratio=self._poisson_ratio,
+                            seabed=self._seabed,
+                        )
+                        dz2 = self.subsidence.evalGeertsma(
+                            base_survey="base",
+                            monitor_survey="monitor",
+                            pos=r2,
+                            youngs_modulus=self._youngs_modulus,
+                            poisson_ratio=self._poisson_ratio,
+                            seabed=self._seabed,
+                        )
+
+                        ezz = (dz2 - dz1) * 10
```

Comment:
What is `ezz`, and why is it multiplied by 10?

Reply:
I've checked the code of `ecl_grav_common_eval_geertsma_kernel` (being used by `subsidence.evalGeertsma`), which is essentially a convolution (over voxel grid of velocities) by applying `ecl_grav_common_eval_geertsma_kernel`. This ends up being weighted sum velocities; therefore it looks like conversion from speed (or its difference) into distance, where `time=10`, but no idea about the units (years? :)). I've contacted the original author about some other weird constants, like the negative shift division, and no reply yet. Wondering if this should be attached.

Reply:
I have to say, I still don't understand what `ezz` is supposed to be an acronym of.. But, yeah, I agree to include a question about this as well. Could be something simple, but nice to know (and likely document)

Reply:
Appending answer from Thomas (quite different from what I thought):

ezz denotes the strain in the vertical direction, obtained from the geomechanical model. The vertical strain ezz is equal to dz/z, where dz is the thickness change within a chosen thickness z. 

Regarding the equation,

`ezz = (dz2 - dz1) * 10,`

I do not remember exactly, but by interpretation I guess that the thickness z is chosen to be 1/10 (given in meter, if dz is given in meter); which means that the strain is calculated every 0.1 m. This seems reasonable.

Reply:
I would find it a bit odd that anything that has to do with resolution of either models is burried so far down into the code. Grid cells in eclipse can be small, so I don't think that limitation exists in libecl. I'm not sure if there's a bound limit in segy file format though, but I doubt it.

I seems reasonable to calculate it every n meter (but I would think every 10 cm is too low though, for a 300m depth reservoir, that would accumulate to 300 000 points in z direction), just don't understand how it just happens to be included here.

Reply:
Considering

> to the fact that Snorre uses reversed convention (and Snorre where first user of this workflow).

Just to make sure, this doesn't have anything to do with Snorre. We don't want field specific quantities in the code.

Reply:
> It seems reasonable to calculate it every n meter (but I would think every 10 cm is too low though, for a 300m depth reservoir, that would accumulate to 300 000 points in z direction - or am I messing something up here?), but I would expect it to depend on the resolutions of the grid cell and the segyfile, rather than a fixed number

Unless we read the paper we won't know well enough about the thickness parameter I reckon. To me it looks more like a constant, which be declared in the beginning of the file, but I wouldn't expose it as a free parameter.